core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -883,7 +906,7 @@ public ExpressionAssert assertThat()
                     .withRepresentation(ExpressionAssert.TYPE_RENDERER);
         }
 
-        public record Result(Type type, Object value) {}
+        public record Result(Type type, Object value, Optional<Expression> expression) {}
```

Comment:
What is the optimal expression?

Reply:
Extracted expression from plan to check how value was produced. For example, if transformation was involved or not.

Reply:
When would it be empty?

Also, this change to QueryAssertions doesn't follow the intent and spirit of that API and abstraction. It should be changed to work with AssertJ like this:


```
assertThat(assertions.expression("..."))
    .matches(<ir expression>)
```